linera-core/src/client/mod.rs

```diff
@@ -1941,11 +1946,11 @@ where
 
             let maybe_blob = {
                 let chain_state_view = self.chain_state_view().await?;
-                chain_state_view
-                    .manager
-                    .get()
-                    .pending_blobs
+                let manager = chain_state_view.manager.get();
+                manager
+                    .locked_published_blobs
                     .get(&blob_id)
+                    .or_else(|| manager.locked_used_blobs.get(&blob_id))
```

Comment:
And this is also unnecessary: We are looking for the _published_ blobs, aren't we?

Reply:
We should create an issue, though: If the locked block is using blobs that we don't have in storage, we need to find another way to send them to the validator.
(Currently the validator wouldn't accept them, I think, because it expects the blobs in `BlockProposal` to be only the published ones?)

Reply:
This whole `maybe_blob` block can be removed I think. Since this will now only be called for a new block proposal, when this `maybe_blob` block is removed then that means this will only fail if someone forgets to call `add_pending_blobs` to add this blob to the `ChainClient`'s pending blobs, which is the expected behavior, is it not?